File: Instabug-CocoaLumberjack/DDInstabugLogger.m
//
//  DDInstabugLogger.m
//  Instabug Demo
//
//  Created by Yousef Hamza on 2/1/17.
//  Copyright © 2017 Instabug. All rights reserved.
//

#import "DDInstabugLogger.h"
#import <Instabug/Instabug.h>

@implementation DDInstabugLogger

+ (instancetype)sharedInstance {
    static DDInstabugLogger *sharedInstance;
    static dispatch_once_t onceToken;

    dispatch_once(&onceToken, ^{
        sharedInstance = [[DDInstabugLogger alloc] init];
    });
    return sharedInstance;
}


- (void)logMessage:(DDLogMessage *)logMessage {
    NSString *logMsg = logMessage->_message;
    if (self->_logFormatter) {
        logMsg = [self->_logFormatter formatLogMessage:logMessage];
    }

    NSUInteger flag = logMessage->_flag;

    switch (flag) {
        case DDLogFlagVerbose:
            [IBGLog logVerbose:logMsg];
            break;
        case DDLogFlagDebug:
            [IBGLog logDebug:logMsg];
            break;
        case DDLogFlagInfo:
            [IBGLog logInfo:logMsg];
            break;
        case DDLogFlagWarning:
            [IBGLog logWarn:logMsg];
            break;
        case DDLogFlagError:
            [IBGLog logError:logMsg];
            break;
        default:
            [IBGLog log:logMsg];
            break;
    }
}

@end

File: Instabug-CocoaLumberjack-Swift/DDInstabugLogger.swift
//
//  InstabugLogger.swift
//  Figure8
//
//  Created by Rotem Itzhak on 12/03/2017.
//  Copyright © 2017 Filios inc. All rights reserved.
//

import Foundation
import CocoaLumberjackSwift
import Instabug

public class InstabugLogger : DDAbstractLogger
{
    public static let sharedInstance = InstabugLogger()
    
    override public func log(message logMessage: DDLogMessage) {
        var message: String? = logMessage.message
            
        let ivar = class_getInstanceVariable(object_getClass(self), "_logFormatter")
        if let formatter = object_getIvar(self, ivar) as? DDLogFormatter {
           message = formatter.format(message: logMessage)
        }
            
        if let msg = message {
            switch logMessage.flag {
            case DDLogFlag.verbose:
                Instabug.logVerbose(msg)
                break
            case DDLogFlag.debug:
                Instabug.logDebug(msg)
                break
            case DDLogFlag.error:
                Instabug.logError(msg)
                break
            case DDLogFlag.info:
                Instabug.logInfo(msg)
                break
                case DDLogFlag.warning:
                Instabug.logWarn(msg)
                break
            default:
                Instabug.ibgLog(msg)
            }
        }
    }
}
